pkg/marker/marker.go

```diff
@@ -0,0 +1,126 @@
+package marker
+
+import (
+	"encoding/json"
+	"fmt"
+	"os/exec"
+	"strings"
+
+	"github.com/golang/glog"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	types "k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+var (
+	clientset kubernetes.Interface
+)
+
+const (
+	resourceNamespace = "ovs-cni.network.kubevirt.io"
+)
+
+type patchOperation struct {
+	Op    string      `json:"op"`
+	Path  string      `json:"path"`
+	Value interface{} `json:"value,omitempty"`
+}
+
+func init() {
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		glog.Fatalf("Error while obtaining cluster config: %v", err)
+	}
+
+	clientset, err = kubernetes.NewForConfig(config)
+	if err != nil {
+		glog.Fatalf("Error building example clientset: %v", err)
+	}
+}
+
+func getAvailableResources() (map[string]bool, error) {
+	outputRaw, err := exec.Command(
+		"ovs-vsctl",
+		"--db", "unix:///host/var/run/openvswitch/db.sock",
+		"list-br",
+	).CombinedOutput()
+	if err != nil {
+		return nil, fmt.Errorf("failed to list OVS bridges: %v", string(outputRaw))
+	}
+
+	availableResources := make(map[string]bool)
+	for _, bridgeName := range strings.Split(strings.TrimSpace(string(outputRaw)), "\n") {
+		availableResources[bridgeName] = true
+	}
+
+	return availableResources, nil
+}
+
+func getReportedResources(nodeName string) (map[string]bool, error) {
+	reportedResources := make(map[string]bool)
+	node, err := clientset.
+		CoreV1().
+		Nodes().
+		Get(nodeName, metav1.GetOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("failed to get node: %v", err)
+	}
+
+	for nodeResourceName, _ := range node.Status.Capacity {
+		splitNodeResourceName := strings.Split(nodeResourceName.String(), "/")
+		if len(splitNodeResourceName) == 2 && splitNodeResourceName[0] == resourceNamespace {
+			reportedResources[splitNodeResourceName[1]] = true
+		}
+	}
+
+	return reportedResources, nil
+}
+
+func Update(nodeName string) error {
+	availableResources, err := getAvailableResources()
+	if err != nil {
+		return fmt.Errorf("failed to list available resources: %v", err)
+	}
+
+	reportedResources, err := getReportedResources(nodeName)
+	if err != nil {
+		return fmt.Errorf("failed to list reported resources: %v", err)
+	}
+
+	patchOperations := make([]patchOperation, 0)
+
+	for reportedResource, _ := range reportedResources {
+		if _, available := availableResources[reportedResource]; !available {
+			patchOperations = append(patchOperations, patchOperation{
+				Op:   "remove",
+				Path: fmt.Sprintf("/status/capacity/%s~1%s", resourceNamespace, reportedResource),
+			})
+		}
+	}
+
+	for availableResource, _ := range availableResources {
+		if _, reported := reportedResources[availableResource]; !reported {
+			patchOperations = append(patchOperations, patchOperation{
+				Op:    "add",
+				Path:  fmt.Sprintf("/status/capacity/%s~1%s", resourceNamespace, availableResource),
+				Value: "1000",
```

Comment:
why 1000? add a comment explaining it (I believe it's because API doesn't support infinite resources but not sure).

Reply:
Done